### PR TITLE
Docs: Created docs content for new read Json feature

### PR DESCRIPTION
**Pull request by mperezfuster**
Created some content about the new feature for read from json.
Please let me know if other sections might need to be updated as well.
Preview: https://mireia-pxf-json.sc2-04-pcf1-apps.oc.vmware.com/pxf/6-1/using/hdfs_json.html
https://mireia-pxf-json.sc2-04-pcf1-apps.oc.vmware.com/pxf/6-1/using/objstore_json.html

### CODE REVIEW

File: docs/content/hdfs_json.html.md.erb [hunk 1]
```diff
@@ -67,6 +67,8 @@ coordinates.values[0]
 coordinates.values[1]
 ```
 
+In some cases, you may need to retrieve all the values of an entire array within a single column. For example, when arrays are not always the same lenght, or the lenght of the array is too long to create a column for each possible array index, or when using nested arrays. For this purpose, you must define the array field as a single `TEXT` column instead.
```
Comment (bradfordb-vmware) [inline docs/content/hdfs_json.html.md.erb:70]:
```suggestion
In some cases, you may want to retrieve all the values of a JSON array within a single column. For example, when arrays are not always the same length, or the length of the array is too long to create a column for each possible array index, or when using nested arrays. For this purpose, you can define the JSON array field as a single `TEXT` column instead.
```

File: docs/content/hdfs_json.html.md.erb [hunk 2]
```diff
@@ -82,7 +84,7 @@ The following table summarizes external mapping rules for JSON data.
...
 |-------------------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
 | Primitive type (integer, float, string, boolean, null) | Use the corresponding Greenplum Database built-in data type; see [Greenplum Database Data Types](https://gpdb.docs.pivotal.io/latest/ref_guide/data_types.html). |
-| Array                         | Use `[]` brackets to identify a specific array index to a member of primitive type.                                                                                            |
+| Array                         | Use `[]` brackets to identify a specific array index to a member of primitive type.<br>Use `TEXT` to retrieve the full content of the array in one column.                                                                                           |
```
Comment (bradfordb-vmware) [inline docs/content/hdfs_json.html.md.erb:87]: > Use `[]` brackets to identify a specific array index

I know this was the original text, but is there any way to indicate that an integer needs to be placed inside the square brackets?

File: docs/content/hdfs_json.html.md.erb [hunk 3]
```diff
@@ -222,12 +220,28 @@ FORMAT 'CUSTOM' (FORMATTER='pxfwritable_import');
 
 Notice the use of `.` projection to access the nested fields in the `user` and `coordinates` objects.  Also notice the use of `[]` to access specific elements of the `coordinates.values[]` array.
 
+If you need to retrieve all values of the entire array within a single column, you must define the array field as `TEXT`:
```
Comment (bradfordb-vmware) [inline docs/content/hdfs_json.html.md.erb:223]:
```suggestion
If you need to retrieve all values of the entire array within a single column, you can define the array field as `TEXT`:
```

This is just a personal style preference, if the prevailing docs convention is to use "must" then I'd stick with that.

File: docs/content/hdfs_json.html.md.erb [hunk 4]
```diff
@@ -222,12 +220,28 @@ FORMAT 'CUSTOM' (FORMATTER='pxfwritable_import');
...
+FORMAT 'CUSTOM' (FORMATTER='pxfwritable_import');
+```
+
 To query the JSON data in the external table:
 
 ``` sql
 SELECT * FROM singleline_json_tbl;
 ```
 
+Note the difference between the two ways of defining the `coordinates.values[]` array: using the first example, you will obtain a column per array element, while with the second example, you will obtain a single column with all the values of the array.
```
Comment (bradfordb-vmware) [inline docs/content/hdfs_json.html.md.erb:243]:
```suggestion
Note the difference between the two ways of defining the `coordinates.values[]` array: using the first example, you will define a column per array element, while with the second example, you define a single column with all the values of the array.
```

**Review by bradfordb-vmware — COMMENTED** [review]
I think we might to add some an extra paragraph with the SQL statements to convert the `TEXT` column into a Postgres array column. I think @denalex has that example in the automation tests and was planning to share with you.

File: docs/content/hdfs_json.html.md.erb (hunk 1 above)
Comment (denalex) [inline docs/content/hdfs_json.html.md.erb:70]: I would think in most cases you would want the whole array. So I'd write about it first and then "in some cases" about picking a single element out of it.

File: docs/content/hdfs_json.html.md.erb (hunk 3 above)
Comment (denalex) [reply docs/content/hdfs_json.html.md.erb:223]: `you must define the array field as `TEXT`:` -- this is mixing the terms -- `fields` are JSON elements, `columns` are Greenplum elements. I'd say 

You can retrieve all values of the array as a single column, defined with one of the following types :`TEXT`, `VARCHAR`, or `BPCHAR`.

**Review by denalex — COMMENTED** [review]
We also need to talk about JSON objects that we can now retrieve into TEXT columns

File: docs/content/hdfs_json.html.md.erb (hunk 2 above)
Comment (mperezfuster) [reply docs/content/hdfs_json.html.md.erb:87]: @lisakowen what do you think? I know there is also reference to this above the table:
"coordinates is an object composed of a text field named type and an array of integers named values. Use [] to identify specific elements of the values array as Greenplum Database external table columns"

Comment (lisakowen) [reply docs/content/hdfs_json.html.md.erb:87]: yes, i think we could use [n].

File: docs/content/hdfs_json.html.md.erb (hunk 1 above)
Comment (dyozie) [inline docs/content/hdfs_json.html.md.erb:70]: Suggest removing "entire" in this sentence.

Also, typo x2:  lenght.

File: docs/content/hdfs_json.html.md.erb (hunk 3 above)
Comment (dyozie) [inline docs/content/hdfs_json.html.md.erb:223]: Same suggestion to remove "entire" here.

**Review by dyozie — CHANGES_REQUESTED** [review]

File: docs/content/hdfs_json.html.md.erb [hunk 5]
```diff
@@ -60,13 +60,18 @@ user.id
 user.location
 ```
 
-`coordinates` is an object composed of a text field named `type` and an array of integers named `values`. Use `[]` to identify specific elements of the `values` array as Greenplum Database external table columns:
+`coordinates` is an object composed of a text field named `type` and an array of integers named `values`. 
+
+In order to retrieve all the values of the `values` array within a single column, define the corresponding Greenplum Database external table column with one of the following types: `TEXT`, `VARCHAR` or `BPCHAR`.
+
+Alternatively, use `[]` to identify specific elements of the `values` array as separate Greenplum Database external table columns:
```
Comment (lisakowen) [inline docs/content/hdfs_json.html.md.erb:67]: <code>[<i>n</i>]</code>

File: docs/content/hdfs_json.html.md.erb [hunk 6]
```diff
@@ -207,7 +208,21 @@ The `hdfs:json` profile supports the following \<custom-option\>s:
...
+  id_str TEXT,
+  "user.id" INTEGER,
+  "user.location" TEXT,
+  "coordinates.values" TEXT
+)
+LOCATION('pxf://data/pxf_examples/singleline.json?PROFILE=hdfs:json')
+FORMAT 'CUSTOM' (FORMATTER='pxfwritable_import');
+```
+
+If you need to identify specific elements of the `user.coordinates` array in different columns, use `[]` brackets to define the JSON array:
```
Comment (lisakowen) [inline docs/content/hdfs_json.html.md.erb:223]: define ...  -> identify the JSON array element

**Review by lisakowen — APPROVED** [review]
looks good, a few minor comments.

File: docs/content/hdfs_json.html.md.erb [hunk 7]
```diff
@@ -82,7 +87,7 @@ The following table summarizes external mapping rules for JSON data.
...
 |-------------------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
 | Primitive type (integer, float, string, boolean, null) | Use the corresponding Greenplum Database built-in data type; see [Greenplum Database Data Types](https://gpdb.docs.pivotal.io/latest/ref_guide/data_types.html). |
-| Array                         | Use `[]` brackets to identify a specific array index to a member of primitive type.                                                                                            |
+| Array                         | Use `TEXT`, `VARCHAR` or `BPCHAR` to retrieve the contents of the array in one column.<br>Use <code>[<i>n</i>]</code> brackets to identify a specific array index to a member of primitive type.<br>                                                                                           |
```
Comment (bradfordb-vmware) [inline docs/content/hdfs_json.html.md.erb:90]:
```suggestion
| Array                         | Use `TEXT`, `VARCHAR`, or `BPCHAR` to retrieve the contents of the array in one column.<br>Use <code>[<i>n</i>]</code> brackets to identify a specific array index to a member of primitive type.<br>                                                                                           |
```

Added the missing serial comma.

File: docs/content/hdfs_json.html.md.erb [hunk 8]
```diff
@@ -60,13 +60,18 @@ user.id
 user.location
 ```
 
-`coordinates` is an object composed of a text field named `type` and an array of integers named `values`. Use `[]` to identify specific elements of the `values` array as Greenplum Database external table columns:
+`coordinates` is an object composed of a text field named `type` and an array of integers named `values`. 
+
+In order to retrieve all the values of the `values` array within a single column, define the corresponding Greenplum Database external table column with one of the following types: `TEXT`, `VARCHAR` or `BPCHAR`.
```
Comment (bradfordb-vmware) [inline docs/content/hdfs_json.html.md.erb:65]:
```suggestion
In order to retrieve all the elements of the `values` array within a single column, define the corresponding Greenplum Database external table column with one of the following types: `TEXT`, `VARCHAR`, or `BPCHAR`.
```

**Review by bradfordb-vmware — APPROVED** [review]

File: docs/content/hdfs_json.html.md.erb [hunk 9]
```diff
@@ -60,13 +60,18 @@ user.id
 user.location
 ```
 
-`coordinates` is an object composed of a text field named `type` and an array of integers named `values`. Use `[]` to identify specific elements of the `values` array as Greenplum Database external table columns:
+`coordinates` is an object composed of a text field named `type` and an array of integers named `values`. 
+
+In order to retrieve all the values of the `values` array within a single column, define the corresponding Greenplum Database external table column with one of the following types: `TEXT`, `VARCHAR`, or `BPCHAR`.
+
```
Comment (denalex) [inline docs/content/hdfs_json.html.md.erb:66]:
``` pre
 coordinates.values
 ```
to give an example on how the property should be referred to when reading the whole array

File: docs/content/hdfs_json.html.md.erb [hunk 10]
```diff
@@ -207,7 +208,21 @@ The `hdfs:json` profile supports the following \<custom-option\>s:
...
+  id_str TEXT,
+  "user.id" INTEGER,
+  "user.location" TEXT,
+  "coordinates.values" TEXT
+)
+LOCATION('pxf://data/pxf_examples/singleline.json?PROFILE=hdfs:json')
+FORMAT 'CUSTOM' (FORMATTER='pxfwritable_import');
+```
+
+If you need to identify specific elements of the `user.coordinates` array in different columns, use `[]` brackets to identify the JSON array:
```
Comment (denalex) [inline docs/content/hdfs_json.html.md.erb:223]:
```suggestion
If you need to identify specific elements of the `coordinates.values` array in different columns, use `[]` brackets to identify the JSON array:
```

File: docs/content/hdfs_json.html.md.erb [hunk 11]
```diff
@@ -238,15 +262,29 @@ CREATE EXTERNAL TABLE multiline_json_tbl(
...
-  "coordinates.values[1]" INTEGER
+  "coordinates.values" TEXT
 )
 LOCATION('pxf://data/pxf_examples/multiline.json?PROFILE=hdfs:json&IDENTIFIER=created_at')
 FORMAT 'CUSTOM' (FORMATTER='pxfwritable_import');
 ```
 
 `created_at` identifies the member name of the first field in the JSON record `record_obj` in the sample data schema.
 
+If you need to identify specific elements of the `user.coordinates` array in different columns, use `[]` brackets to define the JSON array:
```
Comment (denalex) [inline docs/content/hdfs_json.html.md.erb:273]:
```suggestion
If you need to identify specific elements of the `coordinates.values` array in different columns, use `[]` brackets to define the JSON array:
```

File: docs/content/objstore_json.html.md.erb [hunk 12]
```diff
@@ -85,10 +85,25 @@ Refer to [Loading the Sample JSON Data to HDFS](hdfs_json.html#jsontohdfs) and [
...
       "user.location" TEXT,
-      "coordinates.values[0]" INTEGER,
-      "coordinates.values[1]" INTEGER
+      "coordinates.values" TEXT
     )
       LOCATION('pxf://BUCKET/pxf_examples/singleline.json?PROFILE=s3:json&SERVER=s3srvcfg')
     FORMAT 'CUSTOM' (FORMATTER='pxfwritable_import');
     ```
 
+- If you need to identify specific elements of the `user.coordinates` array in different columns, use `[]` brackets to identify the JSON array:
```
Comment (denalex) [inline docs/content/objstore_json.html.md.erb:94]:
```suggestion
- If you need to identify specific elements of the `coordinates.values` array in different columns, use `[]` brackets to identify the JSON array:
```

**Review by denalex — APPROVED** [review]

**Pull request merged**